### ISSUE TITLE
clang-tidy is silent about the terrible problem

### ISSUE THREAD

**Issue by uselessgoddess** [llvm/llvm-project#56886]
As you know, `integer overflow` is an undefined behavior and compilers can use optimizations related to this. 
For example:
```c++
x + 1 > x // is always true 
```
But `clang-tidy` is silent even in the obvious case: https://godbolt.org/z/6x5PYch64

I'm sure this has already been discovered, so please redirect to tracked issue

**Comment by uselessgoddess** [comment]
This warns of unused `else`, but does not give reasons
https://godbolt.org/z/WP4hbGddE

**Comment by uselessgoddess** [comment]
Oh no. This is `*-else-after-result`

**Comment by JonasToth** [comment]
I am having a little trouble to fully grasp what you mean.
Which check do you expect to catch what exactly?

Do you want the potential UB flagged or is it flagged but you don't want it to be flagged?!
Please provide a self contained reproducer (godbolt is enough) and a short `Actual Results` vs `Expected Result` and check that misbehaves. This would help out for understanding and working on it! :)

**Comment by uselessgoddess** [comment]
`Expected Resul`:
Hey bro, are you playing with overflow

`Actual Result`:
silence

This behavior can be very different on some platforms and lead to UB. 
But `clang-tidy` does not warn that for `clang` `x + 1 > x` is always true.

https://godbolt.org/z/TfjKPP9xW

**Comment by njames93** [comment]
While there may be merit in detecting this undefined behaviour, We already have the undefined behaviour sanitizer which will detect these situations.

**Comment by uselessgoddess** [comment]
What is this sanitizer?

**Comment by njames93** [comment]
> What is this sanitizer?
https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html
